statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,704 @@
+        start = self.alpha.shape[0] * (self.beta.shape[0] +
+                                       self.det_coef_coint.shape[0])
+        ret_1dim = self.stderr_params[start:start+self.gamma.size]
+        return ret_1dim.reshape(self.gamma.shape, order="F")
+    
+    @cache_readonly
+    def stderr_det_coef(self):
+        if self.det_coef.size == 0:
+            return self.det_coef  # 0-size array
+        ret1_1dim = self.stderr_params[-self.det_coef.size:]
+        return ret1_1dim.reshape(self.det_coef.shape, order="F")
+
+
+
+    @cache_readonly
+    def tvalues_alpha(self):
+        return self.alpha / self.stderr_alpha
+
+    @cache_readonly
+    def tvalues_beta(self):
+        first_rows = np.zeros((self.r, self.r))
+        last_rows = self.beta[self.r:] / self.stderr_beta[self.r:]
+        return vstack((first_rows,
+                       last_rows))
+
+    @cache_readonly
+    def tvalues_det_coef_coint(self):
+        if self.det_coef_coint.size == 0:
+            return self.det_coef_coint  # 0-size array
+        return self.det_coef_coint / self.stderr_det_coef_coint
+
+    @cache_readonly
+    def tvalues_gamma(self):
+        return self.gamma / self.stderr_gamma
+
+    @cache_readonly
+    def tvalues_det_coef(self):
+        if self.det_coef.size == 0:
+            return self.det_coef  # 0-size array
+        return self.det_coef / self.stderr_det_coef
+
+    
+    @cache_readonly
+    def pvalues_alpha(self):
+        return (1-scipy.stats.norm.cdf(abs(self.tvalues_alpha))) * 2  # student-t
+
+    @cache_readonly
+    def pvalues_beta(self):
+        first_rows = np.zeros((self.r, self.r))
+        tval_last = self.tvalues_beta[self.r:]
+        last_rows = (1-scipy.stats.norm.cdf(abs(tval_last))) * 2  # student-t
+        return vstack((first_rows,
+                       last_rows))
+
+    @cache_readonly
+    def pvalues_det_coef_coint(self):
+        if self.det_coef_coint.size == 0:
+            return self.det_coef_coint  # 0-size array
+        return (1-scipy.stats.norm.cdf(abs(self.tvalues_det_coef_coint))) * 2  # student-t
+
+    @cache_readonly
+    def pvalues_gamma(self):
+        return (1-scipy.stats.norm.cdf(abs(self.tvalues_gamma))) * 2  # student-t
+
+    @cache_readonly
+    def pvalues_det_coef(self):
+        if self.det_coef.size == 0:
+            return self.det_coef  # 0-size array
+        return (1-scipy.stats.norm.cdf(abs(self.tvalues_det_coef))) * 2  # student-t
+
+
+
+    @cache_readonly
+    def var_repr(self):
+        pi = self.alpha.dot(self.beta.T)
+        gamma = self.gamma
+        K = self.K
+        A = np.zeros((self.p, K, K))
+        A[0] = pi + np.identity(K) + gamma[:, :K]
+        A[self.p-1] = - gamma[:, K*(self.p-2):]
+        for i in range(1, self.p-1):
+            A[i] = gamma[:, K*i:K*(i+1)] - gamma[:, K*(i-1):K*i]
+        return A
+
+    def predict(self, steps=5):
+        return forecast(self.y_all.T[-self.p:], self.var_repr, 0, steps)
```

Comment:
great that this can be ireused and works.

Does the current vector_ar VAR handle trends in forecast correctly, i.e. can we get the forecast with added trend or only for the pure VAR part?